accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,199 @@
+We cannot add Hub connection if SCC credentials are defined, they should be mutually excluded.
+
+We can re-use and improve existing ISSv1 database tables to save the needed data.
+
+
+## Hub as a proxy for SCC data
+
+The SUSE Manager server needs a set of metadata to be able to operate. Currently that metadata is provided by SCC directly or, in the case of PAYG, provided by the cloud RMT infrastructure. We should also provide this data in SUSE Manager HUB to be consumed by the peripheral servers.
+The minimal endpoints to be provided are:
+   - "/suma/product_tree.json"
+   - "/connect/organizations/products/unscoped"
+
+On top of this, we should also provide an endpoint for peripherals to send status data needed by SCC (example of this is the minions registered and hardware information). Peripheral servers should send this data to the HUB instead of SCC, and the HUB server should consolidate it and send it to SCC.
+
+## Peripheral software channels creation
+
+We need a mechanism to create the channels in the peripheral servers (vendor and CLM's) in the desired organization. The peripheral channel creation must be done from the HUB server through an API. Since we are making special channel creation (defined next), those API methods should be available to server-to-server communication only.
+
+For each peripheral we should define the organization mapping between HUB organizations and peripheral ones, following the rules (similiar to what we have on ISSv1):
+  - Each Hub organization can only be mapped to one peripheral organizations
+  - Peripheral organization can receive data from several different HUB organizations
+
+From the webUI we should be able to select which channels should be added to each peripheral (vendor or custom). This needed to be saved in a new database table on the HUB side.
+
+Steps needed to create the channels on peripheral:
+   - In the HUB, generate authentication tokens to access the channel(s). Re-use the existing channel access table.
+   - In the peripheral create all needed channels (vendor and custom) point to the HUB server in the rhncontentsource table. Each channel needs to have its own entry.
+   - Minimal set of tables will be populated on the peripheral: rhnchannel,  rhnchannelcontentsource, rhnchannelproduct, rhnchannelfamilymembers, rhnproductname, rhnchannelcloned, suseproductchannel.
+
+
+When creating the channels in the peripheral server the content source will reference the correct HUB repository that provides that channel. Peripheral servers need to be configured with the flag `java.unify_custom_channel_management`, which will synchronize custom channels during the nightly mgr-sync process.
```

Comment:
Can you please provide further details on the methodology for establishing the mapping between channels and Hub repositories. Is it a manual process where users need to define the repository for each newly created channel in the peripheral, or does it operate automatically through predefined heuristics?

Reply:
This should be automatic when creating the channel on the peripheral. It will create an entry in the table rhncontentsource and associate it with the channel. With this association repo-sync will work out of the box without any code change.
I will update the description to make it more clear.

Reply:
@admd I have updated the description. Can you have a look if is clear enought